Allow to assumeRole from ConfigurationProvider (#803)
async-aws/aws: docs/authentication/hard-coded.md

When developing, and debugging, the simplest way to configure the client, is to set the credentials in the
client configuration parameters.

> ## Connect with accessKey / accessKeySecret

Reviewer: ```suggestion
## Connect with access key and secret

```

When developing, and debugging, the simplest way to configure the client, is to set the credentials in the
client configuration parameters.

> ## Connect with access key and secret

Reviewer 2: ```suggestion
## Authenticate with access key and secret
```

]);
```

> ## Connect with accessKey / accessKeySecret + assumeRole
> You can use a specific role (assumeRole) by specify a roleArn.
> You can also specify a roleSessionName.

Reviewer: ```suggestion
## Assume role ARN with access key and secret

You can tell the API client to use specific role by providing a ARN to 
the `roleArn` key. You can also specify a `roleSessionName`.
```

]);
```

> ## Assume role ARN with access key and secret

Reviewer 2: ```suggestion
## Assuming a role
```
